Nil panic during `singularity remote list`

## Version of Singularity:

3.1.0-427.gd3de5fd51

### Expected behavior

Remotes are listed.

### Actual behavior

```
sasha:~$ singularity remote list
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1002c79]

goroutine 1 [running]:
github.com/sylabs/singularity/internal/app/singularity.RemoteList(0xc0000afa10, 0x24, 0xc0000afa40, 0x26, 0x0, 0x0)
	singularity/internal/app/singularity/remote_list.go:70 +0x8c9
github.com/sylabs/singularity/cmd/internal/cli.glob..func49(0x1c4bbe0, 0x1cad628, 0x0, 0x0)
	singularity/cmd/internal/cli/remote.go:142 +0x55
github.com/sylabs/singularity/vendor/github.com/spf13/cobra.(*Command).execute(0x1c4bbe0, 0xc0000a0170, 0x0, 0x0, 0x1c4bbe0, 0xc0000a0170)
	singularity/vendor/github.com/spf13/cobra/command.go:766 +0x2ae
github.com/sylabs/singularity/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1c4ca20, 0x0, 0x0, 0x14)
	singularity/vendor/github.com/spf13/cobra/command.go:850 +0x2bb
github.com/sylabs/singularity/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	singularity/vendor/github.com/spf13/cobra/command.go:800
github.com/sylabs/singularity/cmd/internal/cli.ExecuteSingularity()
	singularity/cmd/internal/cli/singularity.go:136 +0x2e
main.main()
	singularity/cmd/singularity/cli.go:17 +0x20
```

### Steps to reproduce behavior

I was trying to configure remotes and get rid of sylabs-token:

```
singularity remote add library cloud.sylabs.io
singularity remote login library
singularity remote  status library
singularity remote list
```

--- a/cmd/singularity/remote_test.go
+++ b/cmd/singularity/remote_test.go
@@ -338,4 +338,39 @@ func TestRemoteList(t *testing.T) {
 			}
 		}))
 	}
+
+	// Prep config by selecting a remote to default to
+	use := []struct {
+		name   string
+		remote string
+	}{
+		{"useCloud", "cloud"},
+	}
+
+	for _, tt := range use {
+		argv := []string{"remote", "--config", config.Name(), "use"}
+		argv = append(argv, tt.remote)
+		t.Run(tt.name, test.WithoutPrivilege(func(t *testing.T) {
+			if b, err := exec.Command(cmdPath, argv...).CombinedOutput(); err != nil {
+				t.Log(string(b))
+				t.Fatalf("unexpected failure: %v", err)
+			}
+		}))
+	}
+
+	testPass = []struct {
+		name string
+	}{
+		{"PopulatedConfigWithDefault"},
+	}
+
+	for _, tt := range testPass {
+		argv := []string{"remote", "--config", config.Name(), "list"}
+		t.Run(tt.name, test.WithoutPrivilege(func(t *testing.T) {
+			if b, err := exec.Command(cmdPath, argv...).CombinedOutput(); err != nil {
+				t.Log(string(b))
+				t.Fatalf("unexpected failure: %v", err)
+			}
+		}))
+	}
 }

--- a/internal/app/singularity/remote_list.go
+++ b/internal/app/singularity/remote_list.go
@@ -15,6 +15,7 @@ import (
 )
 
 const listLine = "%s\t%s\t%s\n"
+const listLineDefault = "[%s]\t%s\t%s\n"
 
 // RemoteList prints information about remote configurations
 func RemoteList(usrConfigFile, sysConfigFile string) (err error) {
@@ -61,17 +62,17 @@ func RemoteList(usrConfigFile, sysConfigFile string) (err error) {
 	tw := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
 	fmt.Fprintf(tw, listLine, "NAME", "URI", "SYS")
 	for _, n := range names {
-		uri := c.Remotes[n].URI
-		if c.DefaultRemote != "" && c.DefaultRemote == n {
-			n = fmt.Sprintf("[%s]", n)
-		}
-
 		sys := "NO"
 		if c.Remotes[n].System {
 			sys = "YES"
 		}
 
-		fmt.Fprintf(tw, listLine, n, uri, sys)
+		uri := c.Remotes[n].URI
+		if c.DefaultRemote != "" && c.DefaultRemote == n {
+			fmt.Fprintf(tw, listLineDefault, n, uri, sys)
+		} else {
+			fmt.Fprintf(tw, listLine, n, uri, sys)
+		}
 	}
 	tw.Flush()
 	return nil

